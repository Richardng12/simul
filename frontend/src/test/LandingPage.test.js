import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import LandingPage from '../pages/landingPage/landingPage';

const mockStore = configureMockStore();

describe('<LandingPage />', () => {
  let store;
  beforeEach(() => {
    store = mockStore({ profileReducer: { loading: false } });
  });

  it('renders without crashing', () => {
    const tree = shallow(
      <MemoryRouter>
        <Provider store={store}>
          <LandingPage />
        </Provider>
      </MemoryRouter>,
    ).dive();
    expect(tree).toMatchSnapshot();
  });
});
