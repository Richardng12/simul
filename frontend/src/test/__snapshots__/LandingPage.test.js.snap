// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage /> renders without crashing 1`] = `
<div
  className="page"
>
  <div
    className="waveContainer"
  >
    <div
      className="wave outerWave"
    >
      <svg
        data-name="Layer 1"
        height="100%"
        preserveAspectRatio="none"
        viewBox="0 0 223.56 400"
        width="100%"
      >
        <title>
          wave
        </title>
        <path
          d="M192 0c66 83 8 195 8 195s-70 112 2 205H0V0z"
          fill="#7C4D0F"
        />
      </svg>
    </div>
    <div
      className="wave middleWave"
    >
      <svg
        data-name="Layer 1"
        height="100%"
        preserveAspectRatio="none"
        viewBox="0 0 223.56 400"
        width="100%"
      >
        <title>
          wave
        </title>
        <path
          d="M192 0c66 83 8 195 8 195s-70 112 2 205H0V0z"
          fill="#CB7E17"
        />
      </svg>
    </div>
    <div
      className="wave innerWave"
    >
      <svg
        data-name="Layer 1"
        height="100%"
        preserveAspectRatio="none"
        viewBox="0 0 223.56 400"
        width="100%"
      >
        <title>
          wave
        </title>
        <path
          d="M192 0c66 83 8 195 8 195s-70 112 2 205H0V0z"
          fill="#f6a333"
        />
      </svg>
    </div>
  </div>
  <div
    className="contentContainer"
  >
    <div
      className="content"
    >
      <div
        className="title"
      >
        <svg
          data-name="Layer 1"
          height="100%"
          id="prefix__Layer_1"
          preserveAspectRatio="none"
          viewBox="0 0 191.46 53.04"
          width="100%"
        >
          <defs>
            <style>
              .prefix__cls-1{fill:#b9b9b9}
            </style>
          </defs>
          <title>
            header
          </title>
          <path
            className="prefix__cls-1"
            d="M122.55 170l-2.85-.71a20.64 20.64 0 01-5.33-2.1 22.64 22.64 0 01-4.64-3.52l3.86-5.48a17.23 17.23 0 005.56 3.8 17 17 0 006.59 1.26q4.8 0 7.44-2a6.93 6.93 0 002.65-5.82 6.77 6.77 0 00-1.2-4.23 7.11 7.11 0 00-3.1-2.32 29.87 29.87 0 00-4.91-1.39.58.58 0 00-.19 0 .6.6 0 01-.2 0l-.46-.11a41.47 41.47 0 01-7.21-2 10.42 10.42 0 01-4.76-4q-2-2.94-2-8.34a16.6 16.6 0 011.84-8.1 12.11 12.11 0 015.36-5.11 20.39 20.39 0 0113.14-1.16 20.9 20.9 0 014.52 1.78 25 25 0 014.39 3l-3.52 5.69a20.13 20.13 0 00-5-3.22 12.78 12.78 0 00-5-1.07 10.7 10.7 0 00-7 2.09 7.28 7.28 0 00-2.5 5.92 6.18 6.18 0 001.29 4.14 7.64 7.64 0 003.22 2.23 42 42 0 005.36 1.49l.23.06h.22l.34.07.33.07a32.48 32.48 0 016.7 2.25 10.86 10.86 0 014.5 4.2 14.93 14.93 0 011.8 7.92v.07a15.7 15.7 0 01-1.91 8 12.44 12.44 0 01-5.54 5.06 20.17 20.17 0 01-8.77 1.74zM215.93 118.43h5.8v51.08h-6v-38.26l.3 2.25-10.88 28.91h-4.35l-10.87-28.2.3-3v38.26h-6v-51.04H190l13 33.73zM239.18 167.81a14 14 0 01-5.3-6.5 26.21 26.21 0 01-1.82-10.31v-32.6h6.19v32.85q0 5.58 2.43 8.64a9.29 9.29 0 0013.78 0q2.44-3.06 2.44-8.64v-32.82h6.2V151a26.21 26.21 0 01-1.82 10.28 14 14 0 01-5.32 6.5 16.9 16.9 0 01-16.78 0zM272.82 118.43H279v51.08h-6.2zm3.1 44.27h25.27v6.81h-25.27zM173.93 134.28c-1.23-3.43-3.34-5.36-5.39-7.22a17.54 17.54 0 01-4-4.58l-.18-.33a10.45 10.45 0 01-1.39-3.39c-.11-1.06-.82-1.82-1.59-1.76a1.88 1.88 0 00-1.4 2v36.31a7 7 0 00-4.49-1.67c-4.14 0-7.5 3.65-7.5 8.14s3.36 8.15 7.5 8.15 7.49-3.65 7.49-8.15v-35.49a25.13 25.13 0 003.88 4.16c1.8 1.63 3.49 3.17 4.37 5.62a12.5 12.5 0 01.75 4.68 20.3 20.3 0 01-2.63 9.53 2.6 2.6 0 00.14 2.88 1.23 1.23 0 002.11-.19A25.42 25.42 0 00175 141a17.64 17.64 0 00-1.07-6.72z"
            transform="translate(-109.73 -117)"
          />
        </svg>
      </div>
      <p
        className="innerText"
      >
        We believe music is better with friends. Give it a try and start listening. Together.
      </p>
      <div
        className="buttonContainer"
      >
        <a
          href="/login"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text button MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Login to Spotify
            </span>
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
